### PR TITLE
Cbind for `compute_function_tree_root`

### PR DESCRIPTION
**Pull request by dbanks12**
# Description

Cbind for `compute_function_tree_root`

**Blocked by: https://github.com/AztecProtocol/barretenberg/pull/247**

# Checklist:

- [x] I have reviewed my diff in github, line by line.
- [x] Every change is related to the PR description.
- [x] I have [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) this pull request to the issue(s) that it resolves.
- [x] There are no unexpected formatting changes, superfluous debug logs, or commented-out code.
- [x] There are no circuit changes, OR specifications in `/specs` have been updated.
- [x] There are no circuit changes, OR a cryptographer has been assigned for review.
- [x] I've updated any terraform that needs updating (e.g. environment variables) for deployment.
- [ ] The branch has been rebased against the head of its merge target.
- [ ] I'm happy for the PR to be merged at the reviewer's next convenience. (Yes, but **merge blocker PR first**)
- [x] New functions, classes, etc. have been documented according to the doxygen comment format. Classes and structs must have `@brief` describing the intended functionality.
- [x] If existing code has been modified, such documentation has been added or updated.

> **Note**
> If you are updating the submodule, please make sure you do it in its own _special_ PR and avoid making changes to the submodule as a part of other PRs.
> To update a submodule, you can run the following commands:
> ```console
> $ git submodule update --recursive
> ```
> Alternatively, you can select a particular commit in `barretenberg/aztec3` that you wish to point to:
> ```console
> $ cd barretenberg
> $ git pull origin aztec3        # This will point to the latest commit in `barretenberg/aztec3`
> $ git checkout <commit_hash>    # Use this if you wish to point to a particular commit.
> $ cd ..
> $ git add . && git commit -m <commit_msg>
> $ git push
> ```

### CODE REVIEW

File: cpp/src/aztec3/circuits/abis/c_bind.test.cpp [hunk 1]
```diff
@@ -113,11 +114,44 @@ TEST(abi_tests, compute_function_leaf)
...
+    // initialize the vector with its size so that a leaf's data can be copied in (via `seralize_to_buffer`)
+    // (uint256_t here means nothing; it is just used because it is the right size (32 uint8_ts))
+    std::vector<uint256_t> leaves(num_nonzero_leaves);
+
+    // generate some random leaves
+    // insert them into the vector of leaf fields (for direct tree root computation)
+    // insert their serialized form into the vector of 32-bytes chunks/uint256_ts
+    // (to be cast to a single mega uint8_t* buffer and passed to cbind)
+    for (size_t l = 0; l < num_nonzero_leaves; l++) {
+        NT::fr leaf = engine.get_random_uint256();
```
Comment (suyash67) [inline cpp/src/aztec3/circuits/abis/c_bind.test.cpp:143]: You can use `NT::fr::random_element()` to sample a random `fr`.

**Comment by suyash67** [comment]
Looks good to me (just added a very minor comment). Also, nice documentation of the new function.

File: cpp/src/aztec3/circuits/abis/c_bind.test.cpp (hunk 1 above)
Comment (dbanks12) [reply cpp/src/aztec3/circuits/abis/c_bind.test.cpp]: Thanks, updated here: https://github.com/AztecProtocol/aztec3-circuits/pull/65/commits/0384b3c770f58733479562e8ab4eb20a9ebcc9ed

**Comment by dbanks12** [comment]
> "@suyash67: 
> Status: I’ve reviewed it and looks good to me.
> Quick comment: Would it be better if we could templatize TREE_SIZE and LeafPreimage instead of having different compute_<TREE_NAME>_tree_root implementations performing the same logic?"

Done here: https://github.com/AztecProtocol/aztec3-circuits/pull/65/commits/c27e4665add84841d65fbf2aad824427091c5c70

**Review by suyash67 — APPROVED** [review]

**Pull request merged**